cli/azd/pkg/project/ai_helper.go

```diff
@@ -571,6 +552,31 @@ func (a *aiHelper) UpdateTraffic(
 		return nil, err
 	}
 
+	// before moving on, we need to validate the state of the online endpoint to be updated with the
```

Comment:
Do we still need this if given that we filter out the deployment? I'm slightly worried that this will stick in the codebase, and we won't ever know when it's a safe time to remove this.

My preference is to log the bug for the service team and live without mitigation for eventual consistency here (knowing that it would be fixed upstream).

Reply:
It also feels that this could be susceptible to potential concurrent updates to the endpoint that just results in an unexpected failure on our side.

Reply:
@vhvb1989 can you file an issue with https://aka.ms/aistudio/createbug and reference in retry logic for future reference/removal.

Reply:
aistudio issue: https://msdata.visualstudio.com/Vienna/_workitems/edit/3185898